Python/codecs.c

```diff
@@ -50,6 +50,26 @@ int PyCodec_Register(PyObject *search_function)
     return -1;
 }
 
+PyObject *
+_PyCodec_Unregister(PyObject *search_function)
+{
+    PyInterpreterState *interp = PyInterpreterState_Get();
+    if (interp->codec_search_path == NULL) {
+        Py_RETURN_NONE;
+    }
+
+    Py_ssize_t n = PyList_Size(interp->codec_search_path);
+    PyObject *list = PyList_New(0);
+    for (Py_ssize_t i = 0; i < n; i++) {
+        PyObject *item = PyList_GetItem(interp->codec_search_path, i);
+        if (item != search_function) {
+            PyList_Append(list, item);
+        }
+    }
```

Comment:
When the function was requested in 2011, @malemburg explained that there is an issue with the internal codec cache.

Please clear `intepr->codec_search_cache`, but only if the search function is removed.

Since https://docs.python.org/dev/library/codecs.html#codecs.lookup documentation mentions an internal cache, you may have to mention in codecs.unregister() documentation that the function clears the cache if the search function is removed.

Reply:
make sense, updated.